Tasks/Common/AzureRmDeploy-common/azure-arm-rest/azure-arm-app-service.ts

```diff
@@ -204,16 +206,28 @@ export class AzureAppService {
         }
     }
 
-    public async patchApplicationSettings(addProperties: any, deleteProperties?: any): Promise<boolean> {
+    public async patchApplicationSettings(addProperties: any, deleteProperties?: any, formatJSON?: boolean): Promise<boolean> {
         var applicationSettings = await this.getApplicationSettings();
         var isNewValueUpdated: boolean = false;
-        for(var key in addProperties) {
-            if(applicationSettings.properties[key] != addProperties[key]) {
-                tl.debug(`Value of ${key} has been changed to ${addProperties[key]}`);
-                isNewValueUpdated = true;
+        if(formatJSON) {
+            for(var key in addProperties) {
+                if(JSON.stringify(applicationSettings.properties[key]) != JSON.stringify(addProperties[key])) {
```

Comment:
remove code duplication. In if just set value of a variable that you use to compare followed by logic.